jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudConnection.java

```
@@ -262,27 +301,29 @@ public void rollback() {}
     @Override
     public void close() {
         try {
+            if (!shouldCloseChannelWithConnection) {
+                log.warn(
+                        "Called close on a DataCloudConnection, but shouldCloseChannelWithConnection was set to false");
+                return;
+            }
```

This warning doesn't make sense to me. Independent of who has channel ownership we should always ask users to close the connection as a general best practice. If there is _currently_ no other cleanup than closing the channel then we can just silently return.

There is also no reason to do this check inside the try catch, let's hoist it outside

---

This would happen in the case where someone did something like this:

```java
try (DataCloudConnection conn = DataCloudConnection.of(channel, properties, false)) {
```

---

I think we should keep this, it basically is a warning that says "you triggered the AutoCloseable using a `try` but you didn't opt-in to the functionality of the automatic cleanup with `true`".

I did lift the warn out of the try/catch though

---

>I think we should keep this, it basically is a warning that says "you triggered the AutoCloseable using a try but you didn't opt-in to the functionality of the automatic cleanup with true".

Let's make it a debug / trace log. I'm not a fan of issuing a warning when the client is doing everything correctly. The intent looks to be more like a trace / debug hint